tests/linearizability/linearizability_test.go

```diff
@@ -383,58 +430,56 @@ type trafficConfig struct {
 	traffic     Traffic
 }
 
-func watchEventHistory(responses [][]watchResponse) (longest []watchEvent, rest [][]watchEvent) {
+func watchEvents(responses [][]watchResponse) [][]watchEvent {
 	ops := make([][]watchEvent, len(responses))
 	for i, resps := range responses {
 		ops[i] = toWatchEvents(resps)
 	}
-
-	sort.Slice(ops, func(i, j int) bool {
-		return len(ops[i]) > len(ops[j])
-	})
-	return ops[0], ops[1:]
+	return ops
 }
 
-func validateEventsMatch(t *testing.T, longestHistory []watchEvent, other [][]watchEvent) {
-	for i := 0; i < len(other); i++ {
-		length := len(other[i])
+func validateEventsMatch(t *testing.T, histories [][]watchEvent) {
+	longestHistory := longestHistory(histories)
+	for i := 0; i < len(histories); i++ {
+		length := len(histories[i])
 		// We compare prefix of watch events, as we are not guaranteed to collect all events from each node.
-		if diff := cmp.Diff(longestHistory[:length], other[i][:length], cmpopts.IgnoreFields(watchEvent{}, "Time")); diff != "" {
-			t.Errorf("Events in watches do not match, %s", diff)
+		if diff := cmp.Diff(longestHistory[:length], histories[i][:length], cmpopts.IgnoreFields(watchEvent{}, "Time")); diff != "" {
+			t.Error("Events in watches do not match")
```

Comment:
Why did you get rid of prininting the diff ?

Reply:
Problem I found is that this can spam whole console with thousands and thousands of lines of logs. This can obscure some other errors logs.
With reports written to file, it's now possible to do the diff yourself after running the test. 

Will add a recommendation to diff the files manually.